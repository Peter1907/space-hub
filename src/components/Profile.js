import { useSelector } from 'react-redux';
import styles from './Profile.module.css';

export default function Profile() {
  const rockets = useSelector((state) => state.rockets);
  const resRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div className={styles.myProfile}>
      <div className={styles.missions}>
        <h2>Missions</h2>
        <div className={styles.missionList} />
      </div>
      <div className={styles.rockets}>
        <h2>Rockets</h2>
        <div className={styles.rocketList}>
          {resRockets.map((rocket) => (
            <p key={rocket.id}>{rocket.name}</p>
          ))}
        </div>
      </div>
    </div>
  );
}
