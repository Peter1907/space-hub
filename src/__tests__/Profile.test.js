/* eslint-disable*/

import renderer from 'react-test-renderer';
import Profile from '../__mocks__/ProfileMock';

it('renders correctly', () => {
  const tree = renderer
    .create(<Profile />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
