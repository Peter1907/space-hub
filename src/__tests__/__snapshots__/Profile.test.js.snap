// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="myProfile"
>
  <div
    className="missions"
  >
    <h2>
      Missions
    </h2>
    <div
      className="missionList"
    >
      <p>
        Telstar
      </p>
      <p>
        Iridium NEXT
      </p>
    </div>
  </div>
  <div
    className="rockets"
  >
    <h2>
      Rockets
    </h2>
    <div
      className="rocketList"
    />
  </div>
</div>
`;
